doc/wasmoo.rst

```diff
@@ -0,0 +1,46 @@
+.. _wasmoo:
+
+***************************************
+Wasm Compilation With Wasm_of_ocaml
+***************************************
+
+.. TODO(diataxis)
+
+   This is an how-to guide.
+
+Wasm_of_ocaml_ is a compiler from OCaml to WebAssembly (Wasm for
+short). The compiler works by translating OCaml bytecode to Wasm code.
+
+Compiling to Wasm is very similar to compiling to JavaScript. See
+:doc:`jsoo` for more information.
+
+
+Compiling to Wasm
+=================
+
+Dune has full support for building wasm_of_ocaml libraries and executables transparently.
+There's no need to customise or enable anything to compile OCaml
+libraries/executables to Wasm.
+
+To build a Wasm executable, just define an executable as you would normally.
+Consider this example:
+
+.. code:: console
+
+   $ echo 'print_endline "hello from wasm"' > foo.ml
+
+With the following ``dune`` file:
+
+.. code:: dune
+
+  (executable (name foo) (modes js) (js_of_ocaml (submodes wasm)))
+
+And then request the ``.wasm.js`` target:
+
+.. code:: console
+
+   $ dune build ./foo.bc.wasm.js
+   $ node _build/default/foo.bc.wasm.js
```

Comment:
Why do I need to specify the submodes here? Dune could find out what to build for this wasm.js target on its own, isn't it?

Reply:
The documentation has not been updated. That should be the following above:
```
(executable (name foo) (modes wasm))
```
But yes, you can just use `dune build`.

Not that this was just copied from the Js_of_ocaml part of the doc:
https://github.com/ocaml/dune/blob/b4099636c73c28ad002559efab1d7c84285acb2d/doc/jsoo.rst?plain=1#L38-L44